### PR TITLE
Search routes

### PR DESCRIPTION
**Pull request by SmitharyZach**
create the search route and make it dynamic to accept any search term
fix the sign up flow so an access token is created on sign up and sent to the front end
expire access tokens after 12 hours
add a get user route to get a user's info

### CODE REVIEW

File: server/routes/search.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const express = require("express");
+const { Op } = require("sequelize");
+const router = express.Router();
+const db = require("../models");
+const authenticateToken = require("./helper/authenticateToken");
+router.use(express.json());
+
+router.get("/search", authenticateToken, (req, res) => {
+  let serachTerms = req.query;
```
Comment (wizardoutofwater) [inline server/routes/search.js:11]: Looks good! except theres a typo here -- serach ... it still works bc the typo is repeated on the next couple lines but if you've got time  can you fix that?

**Review by wizardoutofwater — COMMENTED** [review]
I dig it!  couple things. 

-  I commented on a typo in a couple lines of search.js.
- is req.user.id  pulled from the token? or do i need to send the user.id in req.body

File: server/routes/search.js (hunk 1 above)
Comment (SmitharyZach) [reply server/routes/search.js]: fixed!

**Review by wizardoutofwater — APPROVED** [review]
Good to merge!

**Pull request merged**